api/v1beta1/flinkcluster_validate.go

```diff
@@ -96,13 +102,29 @@ func (v *Validator) ValidateUpdate(old *FlinkCluster, new *FlinkCluster) error {
 		return nil
 	}
 
-	if !reflect.DeepEqual(new.Spec, old.Spec) {
-		return fmt.Errorf("the cluster properties are immutable")
+	err = v.validateJobUpdate(old, new)
+	if err != nil {
+		return err
+	}
+
+	err = v.ValidateCreate(new)
+	if err != nil {
+		return err
 	}
 
 	return nil
 }
 
+func (v *Validator) validateNamespacedName(old metav1.ObjectMeta, new metav1.ObjectMeta) error {
```

Comment:
This cannot be validated, because if the name changes, it becomes Create, the update validation won't even be hit.